tardis/visualization/tools/convergence_plot.py

```diff
@@ -0,0 +1,360 @@
+            name="Inner<br>Boundary<br>Temperature",
+            row=1,
+            col=1,
+            hovertext="text",
+            marker_color="crimson",
+            mode="lines",
+        )
+
+        fig = fig.update_layout(
+            xaxis=dict(range=[0, self.iterations + 1], dtick=2),
+            xaxis2=dict(
+                matches="x",
+                range=[0, self.iterations + 1],
+                dtick=2,
+            ),
+            xaxis3=dict(
+                matches="x2",
+                title=r"$\mbox{Iteration Number}$",
+                dtick=2,
+            ),
+            yaxis=dict(
+                title=r"$\mbox{T}_{inner}\ [K]$",
+                automargin=True,
+                side="top",
+                tickformat="g",
+                exponentformat="e",
+                tickmode="auto",
+                nticks=4,
+            ),
+            yaxis2=dict(
+                exponentformat="e",
+                title=r"$\mbox{Luminosity}~(erg~sec^{-1})$",
+                title_font_size=13,
+                automargin=True,
+                tickmode="auto",
+                nticks=7,
+            ),
+            yaxis3=dict(
+                exponentformat="e",
+                title=r"$~~\mbox{Residual}\\\mbox{Luminosity(%)}$",
+                title_font_size=12,
+                automargin=True,
+                tickformat="%",
+                tickmode="auto",
+                nticks=4,
+            ),
+            legend_tracegroupgap=0,
+            height=630,
+            hoverlabel_align="right",
+            legend_title_text="Luminosity",
+            hoverlabel_font_color="white",
+            margin=dict(b=25, t=25, pad=0),
+        )
+
+        # allows overriding default layout
+        if hasattr(self, "luminosity_plot_config"):
+            self.override_plot_parameters(fig, self.luminosity_plot_config)
+
+        self.luminosity_plot = fig
+
+    def override_plot_parameters(self, fig, parameters):
+        """
+        Overrides default plot properties.
+        """
+        for key, value in parameters.items():
+            if key == "data":
+                for trace in list(fig.data):
+                    self.override_plot_parameters(trace, value)
+            else:
+                if type(value) == dict:
+                    self.override_plot_parameters(fig[key], value)
+                else:
+                    fig[key] = value
+
+    def build(self, display_plot=True):
+        """
+        Calls the create_plasma_plot and the create_luminosity_plot to build plots.
+        """
+        self.create_plasma_plot()
+        self.create_luminosity_plot()
+        if display_plot:
+            display(
+                widgets.VBox(
+                    [self.plasma_plot, self.luminosity_plot],
+                )
+            )
+
+    def update_plasma_plots(self):
+        """
+        Updates the plasma plots using the data collected using the fetch_data function.
```

Comment:
Again no need to tell `..using the fetch_data function` instead explain what it does